core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -108,23 +108,21 @@ public TableScan appendsAfter(long fromSnapshotId) {
     }
 
     /**
-     * @return list of manifest files to explore for this files metadata table scan
+     * @return iterable of manifest files to explore for this files metadata table scan
      */
-    protected abstract List<ManifestFile> manifests();
+    protected abstract CloseableIterable<ManifestFile> manifests();
 
-    private CloseableIterable<ManifestFile> filterManifests(List<ManifestFile> manifests,
+    private CloseableIterable<ManifestFile> filterManifests(CloseableIterable<ManifestFile> manifests,
```

Comment:
nit: not necessarily related but could you fix the arg alignment in `planFiles` above?

Reply:
Done